plugins/inputs/jolokia2/README.md

```diff
@@ -0,0 +1,98 @@
+# Jolokia2 Input Plugin
+
+The [Jolokia](http://jolokia.org) input plugin collects JVM metrics exposed as JMX MBean attributes
+through the Jolokia REST endpoint and its [JSON-over-HTTP protocol](https://jolokia.org/reference/html/protocol.html).
+
+### Configuration:
+
+```toml
+# Read JMX metrics from a Jolokia REST endpoint
+
+[[inputs.jolokia2]]
+  # default_field_separator = "."
+  # default_field_prefix    = ""
+  # default_tag_prefix      = ""
+
+  # Add agents URLs to query
+  urls = ["http://kafka:8080/jolokia"]
+
+  # Supply a 'path' to collect a simple scalar value called 'Uptime'.
+  [[inputs.jolokia2.metric]]
+    name  = "jvm_runtime"
+    mbean = "java.lang:type=Runtime"
+    paths = ["Uptime"]
+
+  # More complex values can be collected.
+  [[inputs.jolokia2.metric]]
+    name  = "jvm_memory"
+    mbean = "java.lang:type=Memory"
+    paths = ["HeapMemoryUsage", "NonHeapMemoryUsage", "ObjectPendingFinalizationCount"]
+
+  # Use 'mbean' object patterns to create distinct series, and
+  # use 'tag_keys' to specify exactly the keys to add as tags.
+  [[inputs.jolokia2.metric]]
+    name     = "jvm_garbage_collector"
+    mbean    = "java.lang:name=*,type=GarbageCollector"
+    paths    = ["CollectionTime", "CollectionCount"]
+    tag_keys = ["name"]
+
+  # Use 'tag_prefix' to add detail to tag names.
+  [[inputs.jolokia2.metric]]
+    name       = "jvm_memory_pool"
+    mbean      = "java.lang:name=*,type=MemoryPool"
+    paths      = ["Usage", "PeakUsage", "CollectionUsage"]
+    tag_keys   = ["name"]
+    tag_prefix = "pool"
+
+  # Use simple substitutions to alter field prefixes with mbean properties.
+  [[inputs.jolokia2.metric]]
+    name         = "kafka_topic"
+    mbean        = "kafka.server:name=*,topic=*,type=BrokerTopicMetrics"
+    field_prefix = "$1"
+    tag_keys     = ["topic"]
+
+  # In some cases, it makes sense to substitute entire field names for
+  # mbean property keys.
+  [[inputs.jolokia2.metric]]
+    name       = "kafka_log"
+    mbean      = "kafka.log:name=*,partition=*,topic=*,type=Log"
+    field_name = "$1"
```

Comment:
I didn't see a spot where the $1 syntax was explained.  Do these correspond to the `*` in the `mbean` options?